### ISSUE TITLE
Callback for Plotting Bounding Boxes, Segmentation Masks.

### ISSUE THREAD

**Issue by oke-aditya** [PyTorchLightning/pytorch-lightning-bolts#535]
## 🚀 Feature

The next release of torchvision 0.9.0 will pack a few plotting utilities that work with detection and segmentation models.
E.g. [`draw_bounding_boxes`](https://github.com/pytorch/vision/blob/master/torchvision/utils.py#L140), `draw_segmentation_masks`.

We will have merged detection models from torchvision, (FRCNN and RetinaNet (soon)) into bolts. So this can be a nice time to implement a callback.

### Motivation

It would be really nice to a Callback that allows the users to log their predictions to TensorBoard, etc.

### Pitch

I'm not sure how callbacks work (will dive into it) but something like [confused logit callback](https://pytorch-lightning-bolts.readthedocs.io/en/latest/vision_callbacks.html#confused-logit) can work I guess (thanks @akihironitta).

Detection models in their validation mode output the predicted boxes, which are really good to visualize.
Maybe we can do something `on_validation_step_end`?

### Alternatives

I'm not sure if we would leave it to the user for such implementation, but I think it is nice to have.

### Additional context

Wait till next release fo torchvision, these features are still not released :smile:

**Comment by Programmer-RD-AI** [comment]
hi,

I wanted to know why is the draw_bounding_boxes

not using cv2

`
import cv2
xmin,ymin,xmax,ymax = 317,265,556,342
x = xmin
y = ymin
w = xmax - xmin
h = ymax - ymin
crop = img[y:y+h,x:x+w]
plt.imshow(crop)
plt.imshow(cv2.rectangle(img,(x,y),(x+w,y+h),(200,0,0),2))
`

Example notebook https://github.com/Programmer-RD-AI/Microcontroller-Detection-Learning-Detectron2/blob/main/00.ipynb

I just wanted to know I am trying to do this issue.

WIth best regards,
Ranuga

**Comment by oke-aditya** [comment]
Hi @Programmer-RD-AI `draw_bounding_boxes` and `draw_segmentation_masks` or in fact all the utilities written in `torchvision.utils` do not use `cv2`

The major reason is to avoid an extra opencv dependency. PIL works well out of box, which is a needed dependency for torchvision. Also the utils API works really well with the models provided in `torchvision.models`

A simple example for them is available over torchvision gallery. 

https://pytorch.org/vision/stable/auto_examples/plot_visualization_utils.html#sphx-glr-auto-examples-plot-visualization-utils-py

**Comment by roboserg** [comment]
Any progress on this in? Is there a built-in way in PyTorch Lighting, or Flash or Bolts to visualize bboxes from model predictions?
